### PR TITLE
[297] Handle Succession edge in diagrams

### PR DESCRIPTION
**Pull request by jerome-obeo**
Bug: https://github.com/eclipse-syson/syson/issues/297

### CODE REVIEW

File: backend/views/syson-diagram-actionflow-view/src/main/java/org/eclipse/syson/diagram/actionflow/view/ActionFlowViewJavaServiceProvider.java [hunk 1]
```diff
@@ -46,6 +47,7 @@ public List<Class<?>> getServiceClasses(View view) {
                     ViewNodeService.class,
                     ViewLabelService.class,
                     UtilService.class,
+                    ViewEdgeService.class,
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-actionflow-view/src/main/java/org/eclipse/syson/diagram/actionflow/view/ActionFlowViewJavaServiceProvider.java:50]: I would prefer `ActionFlowViewEdgeService` inherits from `ViewEdgeService` instead.

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java [hunk 2]
```diff
@@ -182,4 +200,63 @@ public boolean checkTransitionEdgeTarget(Element source, Element target) {
...
     }
+
+    /**
+     * Succession edge label getter service.
+     *
+     * @param succession
+     *            The succession edge
+     * @return the label of the given {@link Succession}
+     */
+    public String getSuccessionLabel(Succession succession) {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:211]: I think this rather should be in `ViewLabelService`. What do you think about?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java [hunk 3]
```diff
@@ -182,4 +200,63 @@ public boolean checkTransitionEdgeTarget(Element source, Element target) {
...
+        String guardLabel;
+        guardLabel = this.getElementsDefaultInitialDirectEditLabel(guardExpressions, (a, b) -> {
+            return a + GUARD_EXPRESSION_SEPARATOR + b;
+        });
+        return guardLabel;
+    }
+
+    public Element reconnectSourceSuccessionEdge(Succession succession, ActionUsage oldSource, ActionUsage newSource) {
+        succession.getSource().replaceAll(e -> {
+            if (e.equals(oldSource)) {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:245]: Please prefer `Objects.equals(e, oldSource)`

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java [hunk 4]
```diff
@@ -182,4 +200,63 @@ public boolean checkTransitionEdgeTarget(Element source, Element target) {
...
+                return newSource;
+            }
+            return e;
+        });
+        return succession;
+    }
+
+    public Element reconnectTargetSuccessionEdge(Succession succession, ActionUsage oldTarget, ActionUsage newTarget) {
+        succession.getTarget().replaceAll(e -> {
+            if (e.equals(oldTarget)) {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:255]: Please prefer `Objects.equals(e, oldTarget)`

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java [hunk 5]
```diff
@@ -36,10 +43,21 @@
  */
 public class ViewEdgeService {
 
+    /**
+     * The default separator used when printing a set of Guard Expressions for a TransitionUsage label
+     */
+    private static final String GUARD_EXPRESSION_SEPARATOR = "& ";
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:49]: Where did you find in the specification that `&` should be used to separate guard expressions?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java [hunk 6]
```diff
@@ -182,4 +200,63 @@ public boolean checkTransitionEdgeTarget(Element source, Element target) {
...
+                    }
+                    return this.viewLabelService.getDefaultInitialDirectEditLabel(action);
+                })
+                .reduce(reduceOperator)
+                .orElse("");
+    }
+
+    private String getGuardExpressionsDefaultDirectEditLabel(EList<Expression> guardExpressions) {
+        String guardLabel;
+        guardLabel = this.getElementsDefaultInitialDirectEditLabel(guardExpressions, (a, b) -> {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:237]: Please do not split declaration and affect of `guardLabel `.

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java [hunk 7]
```diff
@@ -182,4 +200,63 @@ public boolean checkTransitionEdgeTarget(Element source, Element target) {
...
+
+    /**
+     * Succession edge label getter service.
+     *
+     * @param succession
+     *            The succession edge
+     * @return the label of the given {@link Succession}
+     */
+    public String getSuccessionLabel(Succession succession) {
+        String resultLabel = "";
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:212]: Please prefer a StringBuilder

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java [hunk 8]
```diff
@@ -182,4 +200,63 @@ public boolean checkTransitionEdgeTarget(Element source, Element target) {
...
+        var guardExpressions = succession.getGuardExpression();
+        if (!guardExpressions.isEmpty()) {
+            String guardLabel = this.getGuardExpressionsDefaultDirectEditLabel(guardExpressions);
+            if (!guardLabel.isBlank()) {
+                resultLabel += "[" + guardLabel + "]";
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:217]: Please use constants from `LabelConstants` for open and close brackets

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java (hunk 2 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:211]: Agree. Moved.

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java (hunk 3 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:245]: Done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java (hunk 4 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:255]: Done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java (hunk 5 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:49]: This part of the code comes from the state transition view module (org.eclipse.syson.diagram.statetransition.view.services.StateTransitionViewEdgeService) where the guard expression is also displayed as part of the Transition edge label.
Currently, the state transition view diagram does not share its content with other diagrams (not even with the General view diagram), I have extracted this service in the common module as it should be used in several diagrams.

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java (hunk 6 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:237]: Done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java (hunk 7 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java:212]: Done

Comment (AxelRICHARD) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java]: I wanted a StringBuilder not a StringBuffer. I will rebase the PR change it myself.

**Pull request merged**